pyproteininference/reader.py

```diff
@@ -984,3 +1013,455 @@ def _find_psms_with_alternative_proteins(self, raw_psms):
+            if input_filenames is not None:
+                if isinstance(input_filenames, str):
+                    input_filenames = [input_filenames]
+                input_files.extend(input_filenames)
+        if len(input_files) == 0:
+            raise ValueError("For idXML files, at least one file must be supplied as target, decoy, or combined input.")
+        logger.info(f"Reading input from {','.join([str(x) for x in self.combined_files])}")
+        input_files = self.combined_files
+
+        def _sort_protein_hits_by_selected_score(hits, score_key, fallback_score_key):
+            sorting_key = score_key
+            sorting_direction_reversed = False
+            if not all([hit.metaValueExists(score_key) for hit in hits]):
+                if not all([hit.fallback_score_key(score_key) for hit in hits]):
+                    raise ValueError(
+                        f"Score key {score_key} not found in all hits: "
+                        f"{[x.getSequence().toString() for x in hits if not x.metaValueExists(score_key)]}"
+                    )
+                sorting_key = score_key
+                # if additive score type, reverse the sorting (multiplicative score type still is normal sort)
+                if self.parameter_file_object.psm_score_type == Score.ADDITIVE_SCORE_TYPE:
+                    sorting_direction_reversed = True
+            # presort by target / decoy to maintain deterministic parity in how percolator generic results are handled
+            # (with a preference for targets over decoys there)
+            # @todo check that target_decoy is created for all instances (e.g. pepxml)
+            sorted_hits = sorted(hits, key=lambda x: 1 if x.getMetaValue('target_decoy') == "decoy" else 0, reverse=False)
+            return sorted(sorted_hits, key=lambda x: float(x.getMetaValue(sorting_key)), reverse=sorting_direction_reversed)
+
+        list_of_psm_objects = []
+        peptide_tracker = set()
+        all_sp_proteins = set(self.digest.swiss_prot_protein_set)
+        # We only want to get unique peptides... using all messes up scoring...
+        # Create Psm objects with the identifier, percscore, qvalue, pepvalue, and possible proteins...
+
+        peptide_to_protein_dictionary = self.digest.peptide_to_protein_dictionary
+
+        initial_poss_prots = []
+
+        for input_file in tqdm.tqdm(input_files, desc="Reading Input files", unit=" files"):
+
+            protein_ids = []
+            peptide_ids = []
+
+            file_extension = os.path.splitext(input_file)[1].lower()
+            if file_extension == ".mzid":
+                logger.info(f"Reading input as mzIdentML")
+                pyopenms.MzIdentMLFile().load(input_file, protein_ids, peptide_ids)
+            elif file_extension in (".pepxml", ".pep.xml", ".xml"):
+                logger.info(f"Reading input as pepXML")
+                pyopenms.MzIdentMLFile().load(input_file, protein_ids, peptide_ids)
```

Comment:
@coreyeb does this work? Looks like its trying to read a pepxml with the MzIdentMLFile class